Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryBuilder.java

```diff
@@ -2935,6 +2985,58 @@ private DataType getExpressionDefResultType(ExpressionDef expressionDef) {
         throw new IllegalArgumentException(String.format("Invalid context reference from %s context to %s context.", currentExpressionContext(), expressionDef.getContext()));
     }
 
+    /**
+     * Add an identifier to the deque to indicate that we are considering it for consideration for identifier hiding and
+     * adding a compiler warning if this is the case.
+     * <p/>
+     * For example, if an alias within an expression body has the same name as a parameter, execution would have
+     * added the parameter identifier and the next execution would consider an alias with the same name, thus resulting
+     * in a warning.
+     * <p/>
+     * Exact case matching as well as case-insensitive matching are considered.  If known, the type of the structure
+     * in question will be considered in crafting the warning message, as per the {@link Element} parameter.
+     *
+     * @param identifier The identifier belonging to the parameter, expression, function, alias, etc, to be evaluated.
+     * @param onlyOnce   Special case to deal with overloaded functions, which are out scope for hiding.
+     * @param element    The construct element, for {@link ExpressionRef}.
+     */
+    void pushIdentifierForHiding(String identifier, boolean onlyOnce, Element element) {
+        final MatchType matchType = identifiersToCheckForHiding.stream()
+                .map(innerIdentifier -> {
+                    if (innerIdentifier.equals(identifier)) {
+                        return MatchType.EXACT;
+                    }
+
+                    if (innerIdentifier.equalsIgnoreCase(identifier)) {
+                        return MatchType.CASE_IGNORED;
+                    }
+
+                    return MatchType.NONE;
+                })
+                .filter(innerMatchType -> MatchType.NONE != innerMatchType)
+                .findFirst()
+                .orElse(MatchType.NONE);
+
+        if (MatchType.NONE != matchType && ! onlyOnce) {
+            final String message = String.format("Identifier hiding detected: Identifier for identifiers: %s%s",
```

Comment:
I'm a little concerned about the message here, can we make this something more friendly, like:

The alias identifier X is hiding the parameter identifier X

And only do this for the exact match case (because technically case-insensitive matches are not hiding, right?)

Reply:
My branch now outputs two distinct messages depending on whether the hiding is for an exact or case-sensitive match.